File: src/Web/CombatAnalysis.EnhancedWebApp/combatanalysis.enhancedwebapp.client/src/features/wowGameData/types/dungeon/DungeonModeModel.ts
import type { DungeonModeProgressModel } from './DungeonModeProgressModel';
import type { DungeonModeTypeModel } from './DungeonModeTypeModel';

export type DungeonModeModel = {
    difficulty: DungeonModeTypeModel;
    status: DungeonModeTypeModel;
    progress: DungeonModeProgressModel;
}

File: src/Web/CombatAnalysis.EnhancedWebApp/CombatAnalysis.EnhancedWebApp.Server/Models/WoWGameData/Character/Dungeon/DungeonModeModel.cs
using System.Text.Json.Serialization;

namespace CombatAnalysis.EnhancedWebApp.Server.Models.WoWGameData.Character.Dungeon;

public class DungeonModeModel
{
    [JsonPropertyName("difficulty")]
    public DungeonModeTypeModel Difficulty { get; set; }

    [JsonPropertyName("status")]
    public DungeonModeTypeModel Status { get; set; }

    [JsonPropertyName("progress")]
    public DungeonModeProgressModel Progress { get; set; }
}
